NorthstarDedicatedTest/scriptJson.cpp

```diff
@@ -0,0 +1,670 @@
+#include "pch.h"
+#include "scriptJson.h"
+#include "rapidjson/document.h"
+#include "rapidjson/writer.h"
+#include "rapidjson/stringbuffer.h"
+
+#include "squirrel.h"
+
+#ifdef _MSC_VER
+#undef GetObject // fuck microsoft developers
+#endif
+
+void sq_EncodeJSON_Table(rapidjson::Value* obj, SQTable* sqTable, rapidjson::Document* allocatorDoc);
+void sq_EncodeJSON_Array(rapidjson::Value* obj, SQArray* sqArray, rapidjson::Document* allocatorDoc);
+void serverSq_DecodeJSON_Table(void* sqvm, rapidjson::Value* obj);
+void serverSq_DecodeJSON_Array(void* sqvm, rapidjson::Value* arr);
+
+SQRESULT serverSq_DecodeJSON(void* sqvm)
+{
+	const char* json = ServerSq_getstring(sqvm, 1);
+	rapidjson::Document doc;
```

Comment:
should use `rapidjson_document` instead of all `rapidjson::Document`

Reply:
Why? We already do it like that in modmanager.cpp and all the tutorial I could find do it like this as well. Also if I change it all member function can not be resolved anymore.

Reply:
Because it's the version that uses Source's memory allocator. The goal was to avoid STL one everywhere, and this basically undoes that. I'm aware modmanager.cpp uses that, I told Bob to fix it but he ignored it eventually xD
Most projects have the default rapidjson::Document typedef with standard allocator, since they don't use custom allocators, and yeah in such case its usage changes a little bit, you can look into the places where rapidjson_document is used to see how the other stuff around was changed

Reply:
It's defined in memalloc.h and memalloc.cpp; example usage: https://github.com/R2Northstar/NorthstarLauncher/blob/47ea109be0d73120ddfe6896494e1fce986515b3/NorthstarDedicatedTest/modmanager.cpp#L476

Reply:
Ok thanks for explaining. Is it better now?